checks/remotesettings/certificates_expiration.py

```diff
@@ -0,0 +1,86 @@
+"""
+Signature certificates should expire in a minimum number of days.
+
+The list of collections whose certificate expires too soon is returned, along
+with the expiration date and x5u URL.
+"""
+import asyncio
+import logging
+from datetime import datetime
+
+import cryptography
+import cryptography.x509
+import requests
+from cryptography.hazmat.backends import default_backend as crypto_default_backend
+from poucave.typings import CheckResult
+
+from .utils import KintoClient as Client
+
+
+logger = logging.getLogger(__name__)
+
+
+def fetch_collection_metadata(server_url, entry):
+    client = Client(
+        server_url=server_url, bucket=entry["bucket"], collection=entry["collection"]
+    )
+    return client.get_collection(_expected=entry["last_modified"])["data"]
+
+
+def fetch_certificate_expiration(x5u: str) -> datetime:
+    resp = requests.get(x5u)
+    cert_pem = resp.text.encode("utf-8")
+    cert = cryptography.x509.load_pem_x509_certificate(
+        cert_pem, crypto_default_backend()
+    )
+    return cert.not_valid_after
+
+
+async def run(server: str, min_remaining_days: int) -> CheckResult:
+    loop = asyncio.get_event_loop()
+
+    client = Client(server_url=server, bucket="monitor", collection="changes")
+    entries = client.get_records()
+
+    futures = [
+        loop.run_in_executor(None, fetch_collection_metadata, server, entry)
+        for entry in entries
+    ]
+    results = await asyncio.gather(*futures)
+    entries_metadata = zip(entries, results)
+
+    x5us = list(
+        {
+            metadata["signature"]["x5u"]
+            for metadata in results
+            if "signature" in metadata
+        }
+    )
+    futures = [
+        loop.run_in_executor(None, fetch_certificate_expiration, x5u) for x5u in x5us
+    ]
+    results = await asyncio.gather(*futures)
+    expirations = {x5u: expiration for x5u, expiration in zip(x5us, results)}
+
+    errors = {}
+    for entry, metadata in entries_metadata:
+        cid = "{bucket}/{collection}".format(**entry)
+        try:
+            x5u = metadata["signature"]["x5u"]
+        except KeyError:
+            errors[cid] = {"x5u": None, "expires": None}
+            continue
```

Comment:
This `try` seems a bit excessive. We don't use `try` around `fetch_certificate_expirations`, for instance (which I think is good and correct, because we don't expect that fetch to fail or produce bad data). Besides, it kind of conflates "doesn't have a signature" and "signature doesn't have a certificate". Maybe we should distinguish these?

Reply:
I agree!

We have other checks that will fail nicely about this missing signature issue. I removed the safety check here.